### DISCUSSION TITLE
category:issue  My C/C++ extension can not work

### DISCUSSION THREAD

**Discussion by XuMingxing** [microsoft/vscode-cpptools#8574] [Q&A]
My C/C++ extension can not work when I use vs code to editor a C file. 
    The outline shows "No symbols found in document xxx".  
    And when I pressed F12 or press ctrl+ left click , it always shows "no definition found for xxx".
    I found the C/C+ extension runtime status has som uncaught errors: "Message header must separate key and value using: "

    I need help.

**Comment by Colengms** [comment]
Hi @XuMingxing .  I don't believe there are any known issue that would be causing that behavior.  The issue would seem to be related to your environment, configuration, and/or the source file involved.  Could you provide more information, such as any settings related to the C/C++ extension, and example code that we might use to reproduce the issue?

**Reply by XuMingxing** [reply]
thanks for reply.  I don't know how to reproduce this issue, I have tried to uninstall vscode from my PC and delete the  "C:\Users\xxx\\.vscode" and "C:\Users\xxx\AppData\Roaming\code",  then reinstall vscode and install c/c++ extension, unfortunately this issue still exist. 
my vscode version is  x64-1.63.0.

**Reply by XuMingxing** [reply]
PS: When I try to open the configurations of C/C++ by command, it is not responding
![image](https://user-images.githubusercontent.com/29743580/146313985-0d4365ef-f65c-4ec4-a5d3-63823b969a97.png)

**Reply by Colengms** [reply]
Hi @XuMingxing .  It sounds like the C/C++ extension is failing to start up or is crashing.  Could you enable `"C_Cpp.loggingLevel": "Debug"` and check the output in the C/C++ output channel?  There may be an error indicated there.

Are you using a Ryzen 3000 series processor?  If so, there was a processor bug that could cause this problem, that can be fixed by updating your BIOS (or using our latest insiders build, which you can find here: https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.0-insiders2 ).  That issue is being tracked by: https://github.com/microsoft/vscode-cpptools/issues/8312

**Reply by XuMingxing** [reply]
Hi, I set "C_Cpp.loggingLevel" to "Debug",  but there is not C/C++ channel in the list.  
![image](https://user-images.githubusercontent.com/29743580/146477937-fa3c337b-ea4e-4e4b-a765-a27c88ce9902.png)

My processor is Inter i5-10500,  and I tried the latest insider build,  it does not work too.
![image](https://user-images.githubusercontent.com/29743580/146479194-15175e85-0b16-45ea-8436-00d696198a54.png)

**Reply by Colengms** [reply]
That is very unusual.  If you open the console output (Help -> Toggle Developer Tools, and the 'console' tab), do any errors related to the C/C++ extension appear there?

**Reply by XuMingxing** [reply]
I can not find any errors related to C/C++.
![image](https://user-images.githubusercontent.com/29743580/146491703-23f862ca-d2aa-422b-9005-8b8ecae58562.png)

**Reply by XuMingxing** [reply]
I saved all the log as below
[vscode-app-1639717651369.log](https://github.com/microsoft/vscode-cpptools/files/7732180/vscode-app-1639717651369.log)

**Reply by Colengms** [reply]
Hi @XuMingxing .  There are a lot of errors being reported in that log, and some of them are related to other extensions (xaver.clang-format, cschlosser.doxdocgen).  Do you still see an issue with the C/C++ extension after disabling all other extensions?

Most of the errors appear to be related to Socket connections.  Is there anything unusual about your system, such as restricted network access?

**Reply by XuMingxing** [reply]
Hi, I disabled all of the extensions except C/C++,  the issue still exists. 
![image](https://user-images.githubusercontent.com/29743580/146632173-192a1df4-bf78-4698-af0a-1265aeb8b604.png)
[vscode-app-1639809943301.log](https://github.com/microsoft/vscode-cpptools/files/7738878/vscode-app-1639809943301.log)

I have not found anything unusual about my network, I can surf online without restriction.

**Reply by sean-mcmanus** [reply]
What version of VS Code are you using? Have you tried uninstall/reinstalling the extension? If that doesn't fix it, does the issue still occur when you change C_Cpp.updateChannel to "Default" to go back to using 1.7.1?

**Reply by sean-mcmanus** [reply]
Oh, ignore my comment -- it looks like you've already provided all that info previously.

**Reply by sean-mcmanus** [reply]
It appears the vscode-jsonrpc module is getting a malformed header somehow. You could try asking for help at https://github.com/microsoft/vscode-languageserver-node/issues -- they might know how to get better logging for what the malformed json message is.

**Reply by XuMingxing** [reply]
Unfortunately, I opened a issue in vscode language server, but they did not support any C++ feature, and closed the issue. https://github.com/microsoft/vscode-languageserver-node/issues/874.

They suggest reporting the issue against the owner of the extension.

**Reply by sean-mcmanus** [reply]
@XuMingxing Can you add `"cpptools.trace.server":"verbose"` to your settings.json and look in the "cpptools" logging for the json message that has the bad header?

You or I could also convert this "discussion" to an "issue".

**Reply by XuMingxing** [reply]
Hi, I enabled the trace and saved the log, but I did not find what was wrong, could you help check it, thanks
[cpptools.json.txt](https://github.com/microsoft/vscode-cpptools/files/7804850/cpptools.json.txt)
.

**Reply by XuMingxing** [reply]
I opened an issue for this discussion. 
https://github.com/microsoft/vscode-cpptools/issues/8628

**Reply by sean-mcmanus** [reply]
Unfortunately, the VS Code team is suggesting that you debug the issue yourself: https://github.com/microsoft/vscode-languageserver-node/issues/874#issuecomment-1006379585 . We can't debug it ourselves unless we can somehow get a repro. This is the only report we've had of this issue, so it seems like something unusual is occurring.

**Reply by XuMingxing** [reply]
I tried to reinstall sometimes, but the issue still exist.  if you need debug, can you add some log in your codes and give me a patch, I will  try to reproduce and send the log.

**Reply by sean-mcmanus** [reply]
The VS Code team is suggesting you debug their JavaScript. It can be done via the Help->Toggle Developer Tools option.

I don't believe adding logging to our extension's code would help.

I think the only way to know what is going on is to set a breakpoing in the JavaScript code that is showing the "Message header" error.

**Reply by sean-mcmanus** [reply]
I can't debug it myself because it doesn't repro for me. The error message in the developer console may have a link to the source file and you could set a breakpoint from there.

**Reply by XuMingxing** [reply]
I can find the point of the error message in the source code, but I don't know how to debug it. the console log is below. 
[vscode-app-1641616775985.log](https://github.com/microsoft/vscode-cpptools/files/7832562/vscode-app-1641616775985.log)

**Reply by sean-mcmanus** [reply]
In the developer console, there should be a link on the error: `workbench.desktop.main.js:formatted:285847 [[object Object]]Message header must separate key and value using :
$onExtensionRuntimeError @ workbench.desktop.main.js:formatted:285847
workbench.desktop.main.js:formatted:285848 Error: Message header must separate key and value using :
	at ...\.vscode\extensions\ms-vscode.cpptools-1.7.1\dist\main.js:58801:23` that you can click to get to the javascript and you should be able to set a breakpoint on that line, then try to inspect the value json data to see why what value of the message header is that gives the error. There's some documentation at https://github.com/microsoft/vscode/wiki/How-to-Contribute#debugging (might be better documentation elsewhere).
